pkg/application/cluster_app.go

```diff
@@ -207,17 +211,47 @@ func (c *Cluster) Build() (*BuiltCluster, error) {
 				return builtCluster, err
 			}
 
-			releaseBuilder = releaseBuilder.
-				// Ensure release has a unique name
-				WithPreReleasePrefix("t").WithRandomPreRelease(10).
-				// Set the Cluster App to use
-				WithClusterApp(strings.TrimPrefix(builtCluster.Cluster.App.Spec.Version, "v"), builtCluster.Cluster.App.Spec.Catalog)
-
-			// TODO: Override default App versions if needed
-
-			release, err := releaseBuilder.Build(context.Background())
-			if err != nil {
-				return builtCluster, err
+			overrideReleaseVersion := os.Getenv(env.ReleaseVersion)
+			if overrideReleaseVersion != "" {
+				overrideReleaseCommit := os.Getenv(env.ReleaseCommit)
+				if overrideReleaseCommit == "" {
+					return nil, fmt.Errorf("'%s' was set without also setting '%s'", env.ReleaseVersion, env.ReleaseCommit)
+				}
+
+				// Remove the provider prefix from the release version
+				overrideReleaseVersion = strings.TrimPrefix(overrideReleaseVersion, fmt.Sprintf("%s-", provider))
+
+				release, err = releaseClient.GetReleaseForGitReference(context.Background(), provider, overrideReleaseVersion, overrideReleaseCommit)
```

Comment:
env var name is `E2E_RELEASE_COMMIT`, and this func can accept also branch and tag name. So can we use branch/tag name also for `E2E_RELEASE_COMMIT`?

Reply:
That wont work for the Tekton Pipeline but you could locally I suppose.